prusti-specs/src/specifications/preparser.rs

```diff
@@ -81,6 +80,7 @@ fn error<T>(span: Span, msg: &str) -> syn::Result<T> {
 struct PrustiTokenStream {
     tokens: VecDeque<PrustiToken>,
     source_span: Span,
+    // TODO: can we somehow update the span after popping stuff?
```

Comment:
Would this get us more precise error reporting? I think the span here is for cases where there are no tokens but we still want a span (of the enclosing group, for example).

Reply:
For example when you use the new `parse_rest` function and it encounters additional tokens past what it needed to parse, I was expecting the span to just cover those unconsumed tokens at that point, and I feel like that would be generally useful. For cases where there are no tokens, I'm not sure the span of "everything parsed as prusti tokens" is more useful than just `Span::call_site()` and such.